app/jsx/pages/SearchPage.jsx

```diff
@@ -299,34 +302,50 @@ class FacetFieldset extends React.Component {
         }
       }
     }
-    this.props.handler(event, newQuery, 'pub_year');
+    this.props.handler(event, newQuery, 'pub_year')
+  }
+
+  getFacetNodes = facets => {
+    return facets.map( facet => {
+      var facetItemData = {
+        facetType: this.props.data.fieldName,
+        facet: facet,
+      }
+      return ( <FacetItem key={facet.value} data={facetItemData} query={this.props.query.filters} handler={this.handleChange} /> )
+    })
   }
 
   render() {
-    var facetItemNodes;
-    if (this.props.data.facets) {
-      facetItemNodes = this.props.data.facets.map( facet => {
-        var facetItemData = {
-          facetType: this.props.data.fieldName,
-          facet: facet,
-        }
-        return ( <FacetItem key={facet.value} data={facetItemData} query={this.props.query.filters} handler={this.handleChange} /> )
-      });
+    if (!this.state.deptModalOpen) {
+      var facets, facetItemNodes
+      if (this.props.data.facets) {
+        facets = (this.props.deptModal) ? this.props.data.facets.slice(0, 5) : this.props.data.facets
+        facetItemNodes = this.getFacetNodes(facets)
+      } else {
+        //pub_year
+        facetItemNodes = (
+          <PubYear data={this.props.data} query={this.props.query} handler={this.pubDateChange} />
+        )
+      }
+      return (
+        <details className="c-facetbox" id={this.props.data.fieldName} open={this.props.open}>
+          <summary className="c-facetbox__summary"><span>{this.props.data.display}</span></summary>
+          <div className="facetItems c-checkbox">
+            {facetItemNodes}
+            {this.props.deptModal &&
+              <a href="" onClick={e=>this.setState({deptModalOpen:true})}>show more &raquo;</a>
+            }
+          </div>
+        </details>
+      )
     } else {
-      //pub_year
-      facetItemNodes = (
-        <PubYear data={this.props.data} query={this.props.query} handler={this.pubDateChange} />
+      return (
+        <ModalComp isOpen
```

Comment:
Martin I'd like your help here. The modal is currently only displaying for a millisecond. I'm suspecting this is because the modal should be called from a higher level component?